### PR TITLE
Update reserved names

### PR DESCRIPTION
**Pull request by KellerFuchs**
- [x] Reserve RFC2142, services and email aliases names
- [x] Make the list more readable and move it to a separate file
- [x] Automatically pluralize names

### CODE REVIEW

File: provisor/reserved.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+reserved_usernames = [
+    # Names that might be used for fishing
```
Comment (singlerider) [inline provisor/reserved.py]: Phishing?

File: provisor/reserved.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    ## Network operations
+    'abuse', 'noc', 'security'
+
+    ## Support for specific services
+    'ftp', 'hostmaster', 'news', 'usenet',
+    'uucp', 'postmaster', 'webmaster', 'www'
+]
+
+reserved_usernames += [ name + 's'
```
Comment (singlerider) [inline provisor/reserved.py]: Not Pep8 compliant

File: provisor/utils.py [hunk 3]
```diff
@@ -67,29 +67,22 @@ def validate_pubkey(value):
...
-    #  https://github.com/hashbang/hashbang.sh/blob/master/src/hashbang.sh#L178-191
+    #  https://github.com/hashbang/hashbang.sh/blob/master/src/hashbang.sh#L186-196
     if re.compile(r"^[a-z][a-z0-9]{,30}$").match(value) is None:
         raise ValueError('Username is invalid')
-    if value in reserved_usernames:
+
+    if value in RESERVED_USERNAMES:
         raise ValueError('Username is reserved')
+
     user_exists = True
```
Comment (singlerider) [inline provisor/utils.py]: Why not include `value` in the error message?

Up to you; just a suggestion.

File: provisor/utils.py [hunk 4]
```diff
@@ -67,29 +67,22 @@ def validate_pubkey(value):
...
     try:
         pwd.getpwnam(value)
     except:
         user_exists = False
+
```
Comment (singlerider) [inline provisor/utils.py]: `except` what?
I know this isn't part of the PR, but it's staring me in the face.
Implicit Exception handling ain't cool.

**Review by singlerider — CHANGES_REQUESTED** [review]
At the very least, PEP8 should be adhered to.

File: provisor/reserved.py (hunk 2 above)
Comment (KellerFuchs) [reply provisor/reserved.py]: OK, added a commit, and `pep8` runs cleanly on `reserved.py`.
(And I'm so bloody not making the rest PEP8-compliant in this PR)

File: provisor/utils.py (hunk 4 above)
Comment (KellerFuchs) [reply provisor/utils.py]: I have another branch which removes this code anyhow  ;)

File: provisor/utils.py (hunk 3 above)
Comment (KellerFuchs) [reply provisor/utils.py]: Then I would need to uniformize with the stuff in `hashbang.sh`, and that's kinda out-of-scope for this PR anyhow.

File: provisor/reserved.py (hunk 1 above)
Comment (KellerFuchs) [reply provisor/reserved.py]: Fixed.

**Comment by singlerider** [comment]
I'll open a PR as soon as this is merged to PEP8 all of this.

**Review by singlerider — APPROVED** [review]

**Comment by KellerFuchs** [comment]
@singlerider If you could do the signed merge, that would be super-convenient.

**Pull request merged**